docs/src/pages/guides/minimizing-bundle-size/minimizing-bundle-size.md

```diff
@@ -74,40 +74,7 @@ However, you need to apply the following steps correctly.
 
 #### 1. Configure Babel
 
-Pick one of the following plugins:
-
-- [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) with the following configuration:
-
-  `yarn add -D babel-plugin-import`
-
-  Create a `.babelrc.js` file in the root directory of your project:
-
-  ```js
-  const plugins = [
-    [
-      'babel-plugin-import',
-      {
-        'libraryName': '@material-ui/core',
-        // Use "'libraryDirectory': ''," if your bundler does not support ES modules
-        'libraryDirectory': 'esm',
-        'camel2DashComponentName': false
-      },
-      'core'
-    ],
-    [
-      'babel-plugin-import',
-      {
-        'libraryName': '@material-ui/icons',
-        // Use "'libraryDirectory': ''," if your bundler does not support ES modules
-        'libraryDirectory': 'esm',
-        'camel2DashComponentName': false
-      },
-      'icons'
-    ]
-  ];
-
-  module.exports = {plugins};
-  ```
+You will need to use a babel plugin to rewrite the import path dynamically.
```

Comment:
This page is not only for create-react-app. Why remove this explainer with a generic: You need to figure this out yourself?

Reply:
That isn't what I'm saying at all. I've actually added an explanation of what the babel plugin is doing.

Reply:
I removed the need to show configuration for two different babel plugins. It is confusing and unnecessary. If not 2, how about 10? There is a whole bunch of these import rewrite plugins.

Reply:
Could you explain why you removed the other plugin explainer? What is confusing? Why is it unnecessary?

Reply:
I thought I just did. But let me try again with more words. Maybe some will stick, maybe some won't...:

* There are examples for two plugins that do exactly the same thing. This documentation makes the user do the research to figure out which one they want to use and why... or like most people, would just randomly pick the first one... without doing any research into things. For that reason, I think it is better to bee a bit more opinionated and lead by example... make the choice and make it simple.
* Less code is always better.
* Simpler is almost always better.
* Fewer decisions is almost always better.
* If the user really feels like they want to do the research into another babel plugin to do exactly the same thing, they can do it on their own. MUI doesn't need to provide documentation for every single babel import rewrite plugin out there.
* The JavaScript ecosystem is rife with too many choices and too many ways of doing the same thing. It is a plague. Half the time, these choices all have tradeoffs. A perfect example of this is CRA vs. Parcel. On the outside Parcel looks amazing and much simpler than CRA. But when you bundle your code with Parcel and realize that it is 500kb larger than with CRA, you start to think... oh maybe CRA isn't so bad after all. Oh and Parcel 2 will be better they say, but it doesn't work with babelrc yet so you can't use these docs. It is bleeding edge whack-a-mole.
* Another good example of the plague is something I just cleaned up in the documentation at your request... remove esm. There is 50 ways to bundle modules and they all require too much thought and excessive documentation.
* Finally, I can't believe I have to argue to make something more simple. Even this argument is too complicated.

Reply:
This change in the documentation is highly related to #17015 cc @merceyz. My opinion on it:
- I almost feel like we are undoing a tradeoff that webpack took by building everything in dev mode. What prevents webpack from applying the same logic? But it seems that, because we have more knowledge of our package structure, we can be smarter than webpack.
- This seems to be a generic concern, unrelated to lodash, antd or Material-UI. People could take advantage of one plugin for all their dependencies
- For each kind of problem and constraints, a single solution should be enough. I believe that these 3 plugins overlap. I think that we can only document one.
- Can we focus on CRA in this pull request and leave this concern to #17015?

Reply:
@oliviertassinari Take it or leave it. I'm tired of going in circles with you guys on friggen documentation. I can't imagine what it would be like working on actual code. Ugh. No wonder this ecosystem never advances past 50 half baked ways of doing the same thing. I'll just go back to being a consumer. Peace out.

Reply:
Thanks for the feedback, could we dive in the frustration. What do you mean by going in circle? Do you think publishing our own Babel plugin, and only documenting it is going in the right direction?

Reply:
I think going with whatever results in the best end user experience with the least amount of configuration and documentation, is the best idea. The less I have to know about the inner workings of all this stuff, the better.

Reply:
We will see what we can do about it! :)

Reply:
@oliviertassinari Thanks man. Your attitude has always been great. Which is why in the PR before this one, I made all of your suggested changes, without question. I'm just not a fan of a few of the people you've allowed to surround yourself with. For that reason, I wish you good luck.

Reply:
@lookfirst I think that we should aim at an emotionally safe place for contributors to interact between each other. I understand if you don't want to spend more time on exploring the pros and cons of the different options we have. Core contributors are doing their best, we are all eager to learn and to improve, it's a large part of what OSS is about.